bigquery_etl/format_sql/tokenizer.py

```diff
@@ -706,11 +706,17 @@ class Identifier(Token):
     pattern = re.compile(r"[A-Za-z_][A-Za-z_0-9]*|`(?:\\.|[^\\`])+`")
 
 
+class ProjectIdentifier(Identifier):
+    """Identifier for a GCP project, can contain hyphens unlike Identifier."""
+
+    pattern = re.compile(r"[A-Za-z](?:[A-Za-z_0-9-]*[A-Za-z_0-9])?|`(?:\\.|[^\\`])+`")
```

Comment:
I believe the doubled backslashes are incorrect since this is a raw string.  But since ``[^`]`` should also match dots, I think this slightly simplified version without backslashes should work just as well:
```suggestion
    pattern = re.compile(r"[A-Za-z](?:[A-Za-z_0-9-]*[A-Za-z_0-9])?|`[^`]+`")
```

Reply:
Actually, nevermind, I misinterpreted what that was intending to do.  I thought it was trying to escape the dot, but it's matching/eating potential backslash escape sequences within the backtick-quoted string.  Carry on. 🫡

Reply:
And it looks like that was copied+modified from `Identifier.pattern` above, which it turns out [I wrote](https://github.com/mozilla/bigquery-etl/blame/d81fb331f5acd8658bc2cfdf35106c50249c1f3e/bigquery_etl/format_sql/tokenizer.py#L706). 😅